File: CloudMusic/src/main/java/com/ken/android/CloudMusic/Adapter/DrawerListAdapter.java
package com.ken.android.CloudMusic.Adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import com.ken.android.CloudMusic.R;

import java.util.ArrayList;

/**
 * Created by axnshy on 16/5/26.
 */
public class DrawerListAdapter extends BaseAdapter {


    private static ArrayList<String> drawerlist;
    private static Context context;

    public DrawerListAdapter(Context context,ArrayList<String> drawerlist) {
        this.drawerlist = drawerlist;
        this.context=context;
    }

    @Override
    public int getCount() {
        return drawerlist.size();
    }

    @Override
    public Object getItem(int position) {
        return drawerlist.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        if (drawerlist == null) {
            return null;
        }
        String item = drawerlist.get(position);
        LayoutInflater layoutInflater = LayoutInflater.from(context);


        convertView = layoutInflater.inflate(R.layout.drawer_listview_item, null);

        TextView itemText= (TextView) convertView.findViewById(R.id.drawerlist_title);
        itemText.setText(drawerlist.get(position));
        return convertView;
    }
}

File: SmartPlay/src/main/java/com/luffy/smartplay/ui/adapter/DrawerListAdapter.kt
package com.luffy.smartplay.ui.adapter

import com.luffy.smartplay.R
import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.TextView
import android.widget.BaseAdapter
import android.content.*
import android.view.View
import com.luffy.smartplay.databinding.DrawerListviewItemBinding
import com.luffy.smartplay.ui.base.BaseBindingAdapter
import com.luffy.smartplay.ui.base.BindingViewHolder
import java.util.ArrayList

/**
 * Created by axnshy on 16/5/26.
 */
class DrawerListAdapter(val items: List<String> = mutableListOf()) : BaseBindingAdapter<DrawerListviewItemBinding>() {

    override fun onBindViewHolder(
        holder: BindingViewHolder<DrawerListviewItemBinding>,
        position: Int
    ) {
        super.onBindViewHolder(holder, position)
        holder.holder.drawerlistTitle.text = items[position]
    }

    override fun createViewHolder(
        layoutInflater: LayoutInflater,
        parent: ViewGroup,
        viewType: Int
    ): DrawerListviewItemBinding {
        return DrawerListviewItemBinding.inflate(layoutInflater,parent,false)
    }

    override fun getItemCount(): Int {return items.size
    }
}